src/main/java/bio/terra/service/snapshot/SnapshotService.java

```diff
@@ -359,6 +361,29 @@ public UUID getFirstSourceDatasetIdFromSnapshotId(
         .orElseThrow(() -> new DatasetNotFoundException("Source dataset for snapshot not found"));
   }
 
+  public SnapshotPreviewModel retrievePreview(UUID snapshotId, String tableName, int count) {
+    Snapshot snapshot = retrieve(snapshotId);
+
+    snapshot.getTables().stream()
+        .filter(t -> t.getName().equals(tableName))
+        .findFirst()
+        .orElseThrow(
+            () ->
+                new SnapshotPreviewException(
+                    "No snapshot table exists with the name: " + tableName));
+
+    try {
+      List<Map<String, Object>> values = bigQueryPdao.getSnapshotTable(snapshot, tableName, count);
+
+      SnapshotPreviewModel previewModel = new SnapshotPreviewModel();
+      previewModel.setResult(List.of(values));
+      return previewModel;
```

Comment:
Not a big deal but you could use a fluent style setter here
```suggestion
      return new SnapshotPreviewModel().result(List.of(values));
```

Reply:
I made some changes using ArrayList and used the fluent style too.